src/iarc7_motion/idle_obstacle_avoider.py

```diff
@@ -0,0 +1,97 @@
+        self._tf_listener = tf2_ros.TransformListener(self._tf_buffer)
+
+        with self._lock:
+            self._obstacle_subscriber = rospy.Subscriber("/obstacles", ObstacleArray, self._update_obstacles)
+
+    def _update_obstacles(self, obstacles):
+        with self._lock:
+            self._obstacle_points = []
+            try:
+                # Convert each obstacle into a numpy vector from the quad frame
+                transform = self._tf_buffer.lookup_transform('level_quad', 'map', rospy.Time.now(), self._timeout)
+                for obstacle in obstacles.obstacles:
+                    obstacle_point = PointStamped(header=obstacle.odom.header, point=obstacle.odom.pose.pose.position)
+                    obstacle_point = tf2_geometry_msgs.do_transform_point(obstacle_point, transform)
+                    obstacle_point = np.array([obstacle_point.point.x, obstacle_point.point.y])
+                    # Ignore any obstacle out of the consideration radius
+                    if np.linalg.norm(obstacle_point) <= self._obstacle_radius:
+                        self._obstacle_points.append(obstacle_point)
+            except Exception as ex:
+                rospy.logwarn("ObstacleAvoider Couldn't lookup transform from map to level_quad")
+
+    def get_safest(self, magnitude):
+        with self._lock:
+            regions = []
+            startOpenRegion = None
+            for angle in np.linspace(0, 2*pi, num=self._vector_step_size):
+                vector_is_safe = True
+                new_vector = np.array([cos(angle), sin(angle)])
+                for obstacle in self._obstacle_points:
+                    # Project the obstacle vector onto the flight vector
+                    obstacle_proj_new_vector = (np.dot(new_vector, obstacle)/np.dot(new_vector, new_vector))*new_vector
+
+                    # Don't look at obstacles projected onto the reflection of this vector
+                    if np.sign(obstacle_proj_new_vector[0]) == np.sign(new_vector[0]) and np.sign(obstacle_proj_new_vector[1]) == np.sign(new_vector[1]):
+                        # find distance of the obstacle to its projection on the flight vector
+                        obstacle_to_vector_distance = np.linalg.norm(obstacle - obstacle_proj_new_vector)
+                        # throw out vectors with obstacles that are too close
+                        if obstacle_to_vector_distance <= self._unsafe_obstacle_threshold:
+                            vector_is_safe = False
+                            break
+
+                if vector_is_safe and startOpenRegion is None:
+                    startOpenRegion = angle
+                elif not vector_is_safe and startOpenRegion is not None:
+                    regions.append([startOpenRegion, angle])
+                    startOpenRegion = None
+
+            # Handle ending on a safe region
+            if startOpenRegion is not None:
+                regions.append([startOpenRegion, 2*pi])
+
```

Comment:
I think we also want to merge regions like `[a, 2pi]` and `[0, b]` or otherwise handle that case